### PR TITLE
feat: change viewport css property value automatically when app metrics changed.

### PR DESCRIPTION
**Pull request by andycall**
In this demo, the scrolling list with the red rectangle was styled using CSS viewport units: vh and vw. When the app metrics information changed (keyword popup), the list view's height was automatically updated.

https://github.com/user-attachments/assets/1cb5e9a5-e462-44d0-941a-f8e3a850a9f4

### CODE REVIEW

**Review by looseyi — APPROVED** [review]

File: webf/lib/src/css/values/length.dart [hunk 1]
```diff
@@ -64,6 +64,14 @@ class CSSLengthValue {
...
       }
     }
+
+    if (isViewportSizeRelatedLength()) {
+      renderStyle!.addViewportSizeRelativeProperty();
+    }
+  }
+
+  bool isViewportSizeRelatedLength() {
+    return type == CSSLengthType.VH || type == CSSLengthType.VH;
```
Comment (looseyi) [inline webf/lib/src/css/values/length.dart:74]: here need to change to VW

File: webf/lib/src/css/values/length.dart [hunk 2]
```diff
@@ -64,6 +64,14 @@ class CSSLengthValue {
...
       }
     }
+
+    if (isViewportSizeRelatedLength()) {
+      renderStyle!.addViewportSizeRelativeProperty();
+    }
+  }
+
+  bool isViewportSizeRelatedLength() {
+    return type == CSSLengthType.VH || type == CSSLengthType.VW;
```
Comment (jwxbond) [inline webf/lib/src/css/values/length.dart:74]: type 判断 CSSLengthType.VMIN  CSSLengthType.VMAX , AUTO 这介个都会影响 size , 需要考虑吗？

Comment (andycall) [reply webf/lib/src/css/values/length.dart:74]: Auto 不属于影响屏幕宽度有关的尺寸，会自动计算

Comment (jwxbond) [reply webf/lib/src/css/values/length.dart:74]: OK

**Comment by jwxbond** [comment]
OK

**Review by jwxbond — APPROVED** [review]

**Pull request merged**